dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/webapp/WebApp.java

```diff
@@ -1055,27 +1059,33 @@ public void initialize(WebAppConfiguration config, DeployedModule moduleConfig,
                 }
             }
 
+            
+            // if we're checkpointing, call commonInitializationFinally before the initTaskComplet
+            if (checkpointPhase == CheckpointPhase.APPLICATIONS) {
+                commonInitializationFinally(extensionFactories); // NEVER INVOKED BY
```

Comment:
remove the comment // NEVER INVOKED BY

Reply:
I think we may want to do this for DEPLOYMENT phase also.  That way we can do as much as possible during DEPLOYMENT also as long as no code is executed from the application.  As soon as the first <clinit> is done it will trigger a checkpoint for DEPLOYMENT.